labware-library/src/labware-creator/components/ConditionalLabwareRender.tsx

```diff
@@ -19,74 +20,107 @@ interface Props {
   definition: LabwareDefinition2 | null
 }
 
-const calculateViewBox = (args: {
-  bBox: DOMRect | undefined
-  xDim: number
-  yDim: number
-}): string => {
-  const { bBox, xDim, yDim } = args
-
+const calculateViewBox = (bBox: DOMRect): string => {
   // by-eye margin to make sure there is no visual clipping
   const MARGIN = 5
 
   // calculate viewBox such that SVG is zoomed and panned with the bBox fully in view,
   // in a "zoom to fit" manner, plus some visual margin to prevent clipping
-  return `${(bBox?.x ?? 0) - MARGIN} ${(bBox?.y ?? 0) - MARGIN} ${
-    xDim + MARGIN * 2
-  } ${yDim + MARGIN * 2}`
+  const x = bBox.x - MARGIN
+  const y = bBox.y - MARGIN
+  const xDimension = bBox.width + MARGIN * 2
+  const yDimension = bBox.height + MARGIN * 2
+  return `${x} ${y} ${xDimension} ${yDimension}`
+}
+
+const areBBoxesEqual = (
+  a: DOMRect | undefined,
+  b: DOMRect | undefined
+): boolean => {
+  return isEqual(
+    {
+      x: a?.x,
+      y: a?.y,
+      width: a?.width,
+      height: a?.height,
+    },
+    {
+      x: b?.x,
+      y: b?.y,
+      width: b?.width,
+      height: b?.height,
+    }
+  )
 }
 
 export const ConditionalLabwareRender = (props: Props): JSX.Element => {
+  const { definition } = props
+  return definition === null ? (
+    <Placeholder />
+  ) : (
+    <PopulatedPreview definition={definition} />
+  )
+}
+
+const PopulatedPreview = (props: {
+  definition: LabwareDefinition2
+}): JSX.Element => {
   const { definition } = props
   const gRef = useRef<SVGGElement>(null)
   const [bBox, updateBBox] = useState<DOMRect | undefined>(
```

Comment:
```suggestion
  const [bBox, updateBBox] = useState<DOMRect | null>(null)
```
Kind of a nit, `undefined` should generally be reserved for data external to our API. Doing so adds at least some semantic meaning to the various JS none types. At the very least, it's more internally consistent with how we typically initialize state with a none type value.